test/test_nn.py

```diff
@@ -1721,6 +1721,100 @@ def forward(self, input):
         self.assertEqual(out.get_device(), 0)
         self.assertEqual(out.data, expected_out)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_data_parallel_module_cache(self):
+        net = nn.Sequential(nn.Linear(5, 10), nn.ReLU(), nn.Linear(10, 4)).cuda()
+        dp = nn.DataParallel(net)
+        i = Variable(torch.randn(16, 5).cuda())
+
+        @contextlib.contextmanager
+        def assertCacheFlush():
+            dp(i)
+            self.assertTrue(dp._replicas)
+            yield
+            self.assertFalse(dp._replicas)
```

Comment:
I feel funny saying this, but... "This assert won't get called if the yield raises an exception"! :o) (Yes I know none of these are supposed to raise exceptions today, but what if in the future?! CONSTANT VIGILANCE.)

Reply:
Getting an exception there is a clear failure and it's better to avoid shadowing that fact with an assertion failure

Reply:
If you expect that you might want them to start raising - fix the test.